api/src/controllers/record-utils.js

```diff
@@ -36,39 +35,26 @@ const getLocale = record => {
  *
  * @param {Object} record - data record
  * @param {String|Object} error - error object or code matching key in messages
- *
- * @returns undefined
  */
 const addError = (record, error) => {
   if (!record || !error) {
     return;
   }
 
-  if (_.isString(error)) {
+  if (typeof error === 'string') {
     error = {
       code: error,
-      message: ''
     };
   }
 
-  if (_.findWhere(record.errors, { code: error.code })) {
-    return;
-  }
-
-  const form = record.form && record.sms_message && record.sms_message.form;
-
   if (!error.message) {
-    error.message = config.translate(error.code, getLocale(record));
+    error.message = config.translate(error.code, getLocale(record), error.ctx);
   }
 
-  // replace placeholder strings
-  error.message = error.message
```

Comment:
No reason for us to do hard-coded manual template replacing, `config.translate` supports this natively if you just pass the context.